job_definitions/virus_scan_s3_bucket.yml

```diff
@@ -1,49 +1,27 @@
+{% for environment in ['preview', 'staging', 'production'] %}
 - job:
-    name: "virus-scan-s3-bucket"
-    display-name: "Virus scan an S3 bucket"
+    name: "virus-scan-s3-buckets-{{ environment }}"
+    display-name: "Virus scan S3 bucket(s) - {{ environment }}"
     project-type: freestyle
-    description: "Run the `virus-scan-s3-bucket` script to stream documents from S3 to a deployed (ClamAV) virus scanner."
+    description: "Run the `virus-scan-s3-bucket` script to scan & tag files in buckets via the antivirus-api"
     parameters:
       - string:
-          name: BUCKET
-          description: "The bucket to virus scan (e.g. digitalmarketplace-dev-uploads)"
+          name: BUCKETS
+          description: "The buckets to virus scan, comma separated (e.g. digitalmarketplace-dev-uploads,other-bucket)"
       - string:
           name: PREFIX
           description: "An object prefix to use when filtering files to scan (e.g. g-cloud-9/documents)"
-      - string:
-          name: HOST
-          description: "The exposed host on which clamd is available (e.g. https://clamav.some.url"
-      - string:
-          name: PORT
-          default: "3310"
-          description: "The exposed port on which clamd is available (default: 3310)."
       - string:
           name: SINCE
           description: "A timezone-aware ISO8601 datetime string; if provided, only scan objects uploaded after this point in time (e.g. 2018-01-01T12:00:00Z)."
+      - string:
+          name: CONCURRENCY
+          default: 0
+          description: "Number of concurrent requests to make to Antivirus API. 0 disables concurrency & threading entirely"
       - bool:
           name: DRY_RUN
           default: true
-          description: "A dry run will not bypass retrieving the object from S3 and send an empty file object to ClamAV."
-    publishers:
-      - trigger-parameterized-builds:
-          - project: notify-slack
-            condition: SUCCESS
-            predefined-parameters: |
-              USERNAME=Jenkins
-              JOB=Virus scan S3 bucket
-              ICON=:clean:
-              STATUS=SUCCESS
-              URL=<${BUILD_URL}consoleFull|#${BUILD_NUMBER}>
-              CHANNEL=#dm-release
-          - project: notify-slack
-            condition: UNSTABLE_OR_WORSE
-            predefined-parameters: |
-              USERNAME=Jenkins
-              JOB=Virus scan S3 bucket
-              ICON=:can-of-worms:
-              STATUS=FAILED
-              URL=<${BUILD_URL}consoleFull|#${BUILD_NUMBER}>
-              CHANNEL=#dm-release
```

Comment:
I know we send emails if the AV API finds something bad, but what if the script fails for another reason - bad API key or someone's been fiddling with the script repo. (Agree that we probably don't need the success trigger.)

Reply:
Um, yup, sure, we should probably alert on this.

 - [ ]

Reply:
(or maybe the nightly jobs themselves?)